packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+      (file: File) => !isEqualFileBy(['name', 'type', 'size', 'lastModified'], file, fileToRemove),
+    );
+    setSelectedFiles(selectedFilesWithoutRemoved);
+    setInputStateText(getRemoveSuccessMessage(language));
+
+    if (selectedFilesWithoutRemoved.length > 0) {
+      fileListFocusIndexRef.current = index > 0 ? index - 1 : 0;
+      setInputStateText(getRemoveSuccessMessage(language));
+    } else if (inputRef.current) {
+      inputRef.current.focus();
+      setInputStateText(getNoFilesAddedMessage(language));
+    }
+  };
+
+  const onDragEnter = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(true);
+  };
+
+  const onDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragOverDrop(false);
+  };
+
+  useEffect(() => {
+    if (didMountRef.current && onChange) {
+      onChange(selectedFiles);
+    } else {
+      didMountRef.current = true;
+    }
+    // Clear input value on every change to ensure it triggers a onChange event when files are added
+    resetFileInputValue();
+  }, [selectedFiles, onChange]);
+
+  // Compose aria-describedby attribute
+  const ariaDescribedBy: string = [
+    composeAriaDescribedBy(id, helperTextToUse, errorTextToUse, successText, infoTextToUse),
+    hasFilesSelected && fileListId,
+  ]
+    .filter((text) => !!text)
+    .join(' ');
+
+  return (
+    <>
+      <InputWrapper {...wrapperProps}>
+        <div className={styles.fileInputContainer}>
+          {hasDragAndDrop && (
+            <>
+              <div
+                aria-hidden
+                onDragEnter={onDragEnter}
+                onDragOver={onDragEnter}
+                onDragLeave={onDragLeave}
+                onDrop={(event: React.DragEvent<HTMLDivElement>) => {
+                  const { dataTransfer }: { dataTransfer: DataTransfer } = event;
+                  onDragLeave(event);
+                  onFilesChange(Array.from(dataTransfer.files));
+                }}
+                className={classNames(styles.dragAndDrop, isDragOverDrop && styles.dragAndDropActive)}
+                ref={dropAreaRef}
+              >
+                <div className={styles.dragAndDropLabel}>
+                  <IconUpload aria-hidden />
+                  <span className={styles.dragAndDropLabelText}>{dragAndDrop.label}</span>
+                </div>
+              </div>
+              <div className={styles.dragAndDropHelperText}>{dragAndDrop.helperText}</div>
+            </>
+          )}
+          <div className={styles.fileInputWrapper}>
+            <Button
+              aria-hidden
+              variant="secondary"
+              iconLeft={<IconPlus aria-hidden />}
+              onClick={() => passButtonClickToInput()}
+              disabled={disabled}
+            >
+              {buttonLabel}
+            </Button>
+            <input
```

Comment:
We might want to set `tabIndex=-1` to this input, so that keyboard users cannot focus it. Screenreaders should still be able to reach it.

Reply:
- [ ] set tabIndex=-1 for input to prevent unnecessary focus

Reply:
I need to check this from SiteImprov since we are focusing on the input when files are removed 🤔

Reply:
- [ ] make button unfocusable